// @nrwl/express/src/generators/init/init.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.initSchematic = exports.initGenerator = void 0;
const tslib_1 = require("tslib");
const devkit_1 = require("@nrwl/devkit");
const set_default_collection_1 = require("@nrwl/workspace/src/utilities/set-default-collection");
const node_1 = require("@nrwl/node");
const versions_1 = require("../../utils/versions");
function removeNrwlExpressFromDeps(tree) {
    devkit_1.updateJson(tree, 'package.json', (json) => {
        delete json.dependencies['@nrwl/express'];
        return json;
    });
}
function initGenerator(tree, schema) {
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        set_default_collection_1.setDefaultCollection(tree, '@nrwl/express');
        const initTask = yield node_1.initGenerator(tree, {
            unitTestRunner: schema.unitTestRunner,
            skipFormat: true,
        });
        removeNrwlExpressFromDeps(tree);
        const installTask = devkit_1.addDependenciesToPackageJson(tree, {
            express: versions_1.expressVersion,
            tslib: '^2.0.0',
        }, {
            '@types/express': versions_1.expressTypingsVersion,
            '@nrwl/express': versions_1.nxVersion,
        });
        if (!schema.skipFormat) {
            yield devkit_1.formatFiles(tree);
        }
        return () => tslib_1.__awaiter(this, void 0, void 0, function* () {
            yield initTask();
            yield installTask();
        });
    });
}
exports.initGenerator = initGenerator;
exports.default = initGenerator;
exports.initSchematic = devkit_1.convertNxGenerator(initGenerator);
